plugins/resteasy-http/src/main/java/org/radargun/service/RESTEasyHTTPOperations.java

```diff
@@ -0,0 +1,226 @@
+ */
+public class RESTEasyHTTPOperations implements BasicOperations {
+   private static final Log log = LogFactory.getLog(RESTEasyHTTPOperations.class);
+   private final RESTEasyHTTPService service;
+
+   public RESTEasyHTTPOperations(RESTEasyHTTPService service) {
+      this.service = service;
+   }
+
+   @Override
+   public <K, V> HTTPCache<K, V> getCache(String cacheName) {
+      if (service.isRunning()) {
+         if (cacheName != null && (service.cacheName == null || !service.cacheName.equals(cacheName))) {
+            throw new UnsupportedOperationException();
+         }
+         return new HTTPCache<K, V>();
+      }
+      return null;
+   }
+
+   protected class HTTPCache<K, V> implements BasicOperations.Cache<K, V> {
+
+      @Override
+      public V get(K key) {
+         V result = null;
+         if (service.isRunning()) {
+            String target = buildUrl(service.cacheName, key);
+            try {
+               Invocation get = service.getHttpClient().target(target).request().accept(service.getContentType())
+                     .buildGet();
+               result = decodeByteArray(get.invoke(byte[].class));
+            } catch (NotFoundException e) {
+               log.warn("RESTEasyHTTPOperations.get::Key: " + key + " does not exist in cache: "
+                     + service.cacheName);
+            } catch (Exception e) {
+               throw new RuntimeException("RESTEasyHTTPOperations::get request threw exception: " + target, e);
+            }
+         }
+         return result;
+      }
+
+      @Override
+      public boolean containsKey(K key) {
+         if (service.isRunning()) {
+            return get(key) != null;
+         }
+         return false;
+      }
+
+      @Override
+      public void put(K key, V value) {
+         if (service.isRunning()) {
+            try {
+               String target = buildUrl(service.cacheName, key);
+               Invocation put = service.getHttpClient().target(target).request().accept(service.getContentType())
+                     .buildPut(Entity.entity(encodeObject(value), service.getContentType()));
+               Response response = put.invoke();
+               int status = response.getStatus();
+               response.close();
+               if (status != Status.OK.getStatusCode()) {
+                  throw new RuntimeException("RESTEasyHTTPOperations.put::Unexpected HttpStatus: " + status + " "
+                        + response.getStatusInfo().getReasonPhrase() + " for request " + target);
+               }
+            } catch (IOException e) {
+               throw new RuntimeException(e);
+            } catch (NullPointerException e) {
+               throw new RuntimeException(e);
+            } catch (IllegalArgumentException e) {
+               throw new RuntimeException(e);
+            }
+         }
+      }
+
+      @Override
+      public V getAndPut(K key, V value) {
+         // Not Atomic
```

Comment:
I'm not sure I am following you here. You think the put should return the previous value as an entity?